app/pages/settings/email.jsx

```diff
@@ -0,0 +1,289 @@
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!this.state.page === prevState.page) {
+      this.getProjectForPreferences();
+    }
+  }
+
+  getTalkPreferences() {
+    talkClient.type('subscription_preferences').get()
+    .then((preferences) => {
+      this.setState({
+        talkPreferences: preferences
+      });
+    });
+  }
+
+  getProjectForPreferences(user) {
+    return user.get('project_preferences', {
+      page: this.state.page,
+      sort: 'display_name'
+    })
+    .then((preferences) => {
+      if (preferences.length > 0) {
+        const projectIDs = preferences.map(pref => pref.links.project);
+        apiClient.type('projects').get({
+          id: projectIDs
+        })
+        .then((projects) => {
+          this.setState({
+            meta: preferences[0].getMeta(),
+            projectPreferences: preferences,
+            projects
+          });
+        });
+      }
+    })
+    .catch((error) => {
+      console.log('Something went wrong. Error: ', error);
+    });
+  }
+
+  handleProjectPreferenceChange(preference, event) {
+    preference.update({
+      email_communication: !!event.target.checked
+    })
+    .save()
+    .then(updatedPref => this.setState(updatedPref));
+  }
+
+  handleTalkPreferenceChange(preference, event) {
+    preference.update({
+      email_digest: event.target.value
+    })
+    .save()
+    .then(updatedPref => this.setState(updatedPref));
+  }
+
+  nameOfPreference(preference) {
+    const options = {
+      participating_discussions: 'When discussions I\'m participating in are updated',
+      followed_discussions: 'When discussions I\'m following are updated',
+      mentions: 'When I\'m mentioned',
+      group_mentions: 'When I\'m mentioned by group (this.admins, this.team, etc)',
+      messages: 'When I receive a private message',
+      started_discussions: 'When a discussion is started in a board I\'m following'
+    };
+    return options[preference.category];
+  }
+
+  renderProjectPreferences() {
+    const { projects, projectPreferences } = this.state;
+    return projectPreferences.map((projectPreference, i) => {
+      if (projects[i]) {
+        return (
+          <tr key={projectPreference.id}>
+            <td>
+              <input
+                type="checkbox"
+                name="email_communication"
+                checked={projectPreference.email_communication}
+                onChange={this.handleProjectPreferenceChange.bind(this, projectPreference)}
+              />
+            </td>
+            <td>
+              {projects[i].display_name}
+            </td>
+          </tr>
+        );
+      }
+    });
+  }
+
+  renderPagination() {
+    const { meta, page } = this.state;
+    if (meta) {
+      return (
+        <nav className="pagination">{'Page'}
+          <select
+            value={page}
+            disabled={meta.page_count < 2}
+            onChange={e => this.setState({ page: e.target.value })}
```

Comment:
This changes the page number in component state, but that change doesn't trigger an API request for that page of projects/project preferences.

Reply:
Yeah if block in `componentDidUpdate` is not executed, so I might have screwed up the condition conversion. Also I think two pages of projects seem too few. I have 5 in production. 🤔

Reply:
Surely enough, I did. It should be `!(this.state.page === prevState.page)` instead of `!this.state.page === prevState.page`